src/governance.cpp

```diff
@@ -921,37 +930,17 @@ bool CGovernanceObject::IsValidLocally(const CBlockIndex* pindex, std::string& s
 
 CAmount CGovernanceObject::GetMinCollateralFee()
 {
-    CAmount nMinFee = 0;
     // Only 1 type has a fee for the moment but switch statement allows for future object types
     switch(nObjectType) {
-    case GOVERNANCE_OBJECT_PROPOSAL:
-        nMinFee = GOVERNANCE_PROPOSAL_FEE_TX;
-        break;
-    case GOVERNANCE_OBJECT_TRIGGER:
-        nMinFee = 0;
-        break;
-    default:
-      {
-        std::ostringstream ostr;
-        ostr << "CGovernanceObject::GetMinCollateralFee ERROR: Unknown governance object type: " << nObjectType;
-        throw std::runtime_error(ostr.str());
-      }
+        case GOVERNANCE_OBJECT_PROPOSAL:    return GOVERNANCE_PROPOSAL_FEE_TX;
+        default:                            return 0;
```

Comment:
Yeah, I guess that would work.

Note that the intent here was never to explicitly validate the object type, except to the extent needed to determine the minimum collateral, so I'm not sure a check in IsValidLocally is really needed.  

Logically the validity of the governance object depends on the object type only via the collateral fee.  The object type is valid to the extent that a valid collateral fee is known for it. So I think the way I implemented it actually made sense.

In practice when we add more object types, it will probably be best to have this check in only one place (ie. when the collateral is computed).